src/ros_filter.cpp

```diff
@@ -90,6 +90,39 @@ namespace RobotLocalization
     topicSubs_.clear();
   }
 
+  template<typename T>
+  void RosFilter<T>::reset()
+  {
+    // Get rid of any initial poses (pretend we've never had a measurement)
+    initialMeasurements_.clear();
+    previousMeasurements_.clear();
+    previousMeasurementCovariances_.clear();
+
+    // Clear the measurement queue.
+    // This prevents us from immediately undoing our reset.
+    while (!measurementQueue_.empty() && ros::ok())
+    {
+      measurementQueue_.pop();
+    }
+
+    ros::getGlobalCallbackQueue()->clear();
```

Comment:
Hi. I'm trying to figure out why it is exiting itself prematurely. It shouldn't do that at all. How did you figure it out?